scripts/run_upgrade_matrix.sh

```
@@ -15,10 +15,50 @@ REQUIRED_NAMESPACES=("redhat-ods-operator" "redhat-ods-applications")
+TEST_DIR="${LOG_DIR}/opendatahub-tests"
+mkdir -p "$LOG_DIR"
+
+# Set up logging
+exec 1> >(tee -a "${LOG_DIR}/upgrade-matrix-$(date +%Y%m%d%H%M).log")
+exec 2>&1
+
+# Logging function
+log() {
+    local level=$1
+    local message=$2
+    local timestamp=$(date '+%Y-%m-%d %H:%M:%S')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)